wp-stage-switcher.php

```diff
@@ -49,29 +48,29 @@ public function admin_bar_stage_switcher($admin_bar) {
       return;
     }
 
-    $stages = maybe_unserialize(ENVIRONMENTS);
-    $current_stage = WP_ENV;
+    $this->stages = maybe_unserialize(ENVIRONMENTS);
+    $subdomain_multisite = is_multisite() && is_subdomain_install();
 
-    foreach($stages as $stage => $url) {
-      if ($stage === $current_stage) {
+    $admin_bar->add_menu([
+      'id'     => 'environment',
+      'parent' => 'top-secondary',
+      'title'  => ucwords(WP_ENV),
+      'href'   => '#',
+      'meta'   => [
+        'class' => 'environment-' . sanitize_html_class(strtolower(WP_ENV)),
+      ],
+    ]);
+
+    foreach ($this->stages as $stage => $url) {
+      if ($stage === WP_ENV) {
         continue;
       }
 
-      if (is_multisite() && defined('SUBDOMAIN_INSTALL') && SUBDOMAIN_INSTALL && !is_main_site()) {
-        $url = $this->multisite_url($url) . $_SERVER['REQUEST_URI'];
-      } else {
-        $url .= $_SERVER['REQUEST_URI'];
+      if ($subdomain_multisite) {
+        $url = $this->multisite_url($url);
       }
 
-      $admin_bar->add_menu([
-        'id'     => 'environment',
-        'parent' => 'top-secondary',
-        'title'  => ucwords($current_stage),
-        'href'   => '#',
-        'meta'   => [
-          'class' => 'environment-' . sanitize_html_class(strtolower($current_stage)),
-        ],
-      ]);
+      $url = apply_filters('bedrock/stage_switcher_url', rtrim($url, '/') . $_SERVER['REQUEST_URI'], $url, $stage);
```

Comment:
Should the filter name just be the plugin name exactly? So either `roots/wp-stage-switcher` or `wp-stage-switcher`. I don't think `bedrock` should be part of it?

Reply:
I was using the format of the existing `bedrock/stage_switcher_visibility` filter, but I'm happy to change it

Reply:
Ah sorry, forgot about that. Okay this works 👍